pbf/api.py

```diff
@@ -73,18 +74,27 @@ def get_ip(request):
     else:
         return request.META["REMOTE_ADDR"]
 
+class ApiKey(APIKeyHeader):
+    param_name = "X-API-Key"
+    def authenticate(self, request, key):
+        secret = os.getenv('CUSTOM_API_KEY', None)
+        if secret is None:
+            return False
+        if key == secret:
+            return key
+
+header_key = ApiKey()
+
 def ip_whitelist(request):
-    ip = str(os.environ['OWN_IP'])
-    if ip is None:
-        ip = "127.0.0.1"
+    ip = str(os.getenv('OWN_IP', '127.0.0.1'))
     if get_ip(request) == ip:
         return ip
 
 @api.get("/ip", auth=ip_whitelist)
```

Comment:
my read of this endpoint is it's there to help test in development whether auth is working. Wonder if it's worth also making one for the API key?

Reply:
Is this not soley being used for the truthy callable the Ninja custom auth uses? I saw that it was utilized in the `/ip` route but I don't know if having a similar call for keys is a good idea? That seems like it'd be prone to brute force attack... maybe something like `/api-key` can be used to tell you IF an api key is able to be expected???

Reply:
Yeah, the `/headerkey` endpoint would be neither more nor less brute-forceable than the `game_link` one. It does make me wonder about what sorts of things the site should support. So this is desired in cases where the bot's IP may not be known. In such cases, is it at least known to be in some range? If so, maybe that could be used in some way?

Reply:
Hmm... 🤔
I can work on letting the known IP function take CIDR notation so it could accept a swath of addresses marked as good and still make it backward compatible with non-masked IPs (just shove `/32` on the end for single addresses). Using RFC1918 ranges would probably be a safe starting value for this.

Reply:
See if you like this implementation a bit better?